### PR TITLE
Fixed computed property on arrays in Select component

### PR DESCRIPTION
**Pull request by thangdinh**
@Dylnuge @igillis @azirbel

### CODE REVIEW

**Comment by igillis** [comment]
nice -- hopefully this will fix some of the wonkiness that occurs when `content` is an array of scalar values

one nitpick: could we standardize on one of `.[]` or `.@each`? I don't feel strongly, but `.@each` probably makes it easier to extend later to `.@each.foo`.

**Comment by Dylnuge** [comment]
It should be `.[]` if you're not observing anything beneath it, for performance reasons.

File: src/select.coffee [hunk 1]
```diff
@@ -231,7 +231,7 @@ Ember.AddeparMixins.ResizeHandlerMixin, Ember.Widgets.KeyboardHelper,
   # in the textbox
   preparedContent: Ember.computed ->
     if @get('sortLabels') then @get('sortedFilteredContent') else @get('filteredContent')
-  .property 'sortLabels', 'filteredContent', 'sortedFilteredContent'
+  .property 'sortLabels', 'filteredContent.[]', 'sortedFilteredContent.[]'
```
Comment (Dylnuge) [inline src/select.coffee]: Not your fault, but prototype extensions shouldn't be used in this repo.

File: src/select.coffee [hunk 2]
```diff
@@ -350,7 +350,7 @@ Ember.AddeparMixins.ResizeHandlerMixin, Ember.Widgets.KeyboardHelper,
       (@get('groupedContent') or []).filter (item) ->
         not Ember.get(item, 'isGroupOption')
     )
-  .property 'groupedContent'
+  .property 'groupedContent.[]'
```
Comment (Dylnuge) [inline src/select.coffee]: Same thing

File: src/select.coffee [hunk 3]
```diff
@@ -363,7 +363,7 @@ Ember.AddeparMixins.ResizeHandlerMixin, Ember.Widgets.KeyboardHelper,
       index = content.indexOf value
       @setHighlightedIndex index, yes
     value
-  .property 'selectableOptions', 'highlightedIndex'
+  .property 'selectableOptions.[]', 'highlightedIndex'
```
Comment (Dylnuge) [inline src/select.coffee]: And here

Comment (Dylnuge) [inline src/select.coffee]: Actually they're pretty much everywhere. @igillis how does this work properly?

Comment (igillis) [inline src/select.coffee]: Coffeescript magic
```
foo: Ember.computed ->
  console.log 'bar'
.property 'baz'
```
compiles to 
```
foo: Ember.computed(function() {
  return console.log('bar');
}).property('baz')
```

Comment (igillis) [inline src/select.coffee]: and unlike `Ember.observer`, `Ember.computed` does not return just a function.

Comment (Dylnuge) [inline src/select.coffee]: Ah! That's the part that threw me.

**Comment by cyril-sf** [comment]
It seems like we're basically trying to make Ember.Select promise aware. I'm not sure if this is what we really want.

**Comment by cyril-sf** [comment]
This is maybe the real culprit https://github.com/Addepar/ember-widgets/blob/thang/selectComputedProperty/src/select.coffee#L258

**Comment by igillis** [comment]
@cyril-sf why do you say we're trying to make it promise aware? `PromiseArray` should act just like a normal array once it's resolved, no?

At least in the example @thangdinh was showing me, `contentProxy` was being overridden, so I don't think that's the issue.

@thangdinh maybe you can show @cyril-sf the same code you showed me and see if he has any ideas.

**Comment by thangdinh** [comment]
Thanks @igillis . I emailed @cyril-sf :)

**Comment by thangdinh** [comment]
Thanks @igillis for helping me writing tests.
@cyril-sf

**Comment by cyril-sf** [comment]
Tests look good :+1: 

LGTM

@igillis My initial comment about promises was irrelevant ( I mixed up 2 PRs I was tagged on )

I still feel we shouldn't have to add `'filteredContent.[]'` and `'filteredContent'`. But as this change should be harmless, it's ok for me.

**Comment by thangdinh** [comment]
Merged

**Pull request closed without merging**